### ISSUE TITLE
Crash on opening .gz inside .tar.gz

### ISSUE THREAD

**Issue by kuraga** [ib/xarchiver#88]
Crash on opening `f.gz` inside [a.tar.gz](https://github.com/ib/xarchiver/files/3985359/a.tar.gz):

```
(exo-open:22788): dbind-WARNING **: 21:37:32.074: Couldn't register with accessibility bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
xarchiver: xdgmimecache.c:473: cache_glob_node_lookup_suffix: Assertion «character != 0» failed.
Segmentation fault
```

xarchiver 0.5.4.14 on Gentoo

**Comment by ib** [comment]
No chance without an archive (download) the error can be reproduced with.

What does `xarchiver -i` say? Why is `exo-open` involved?

**Comment by kuraga** [comment]
> No chance without an archive (download) the error can be reproduced with.

Seems like my archive wasn't special so added a model example above.

> Why is `exo-open` involved?

Is it used to opening files inside archives, isn't it?

**Comment by ib** [comment]
No problem here.

>> Why is exo-open involved?

> It is used to opening files inside archives, isn't it?

Well, `xarchiver -i` should tell which program is used to handle gzip.

**Comment by kuraga** [comment]
> What does `xarchiver -i` say?

```
gzip:
  /usr/bin/gzip
```

@ib, was it reproduced by you with archive I've added at topic-start-comment?

**Comment by ib** [comment]
What do `file /usr/bin/gzip` and `/usr/bin/gzip -V` say?

> @ib, was it reproduced by you with archive I've added at topic-start-comment?

Yes, of course.

**Comment by kuraga** [comment]
```
$ file /usr/bin/gzip
/usr/bin/gzip: symbolic link to pigz
$ /usr/bin/gzip -V
pigz 2.4
```

**Comment by ib** [comment]
Okay, pigz crashes.

If pigz isn't fully compatible with gzip, I won't support it. I'll have to analyze why there is a crash, though.

**Comment by ib** [comment]
Support has been added.

**Comment by kuraga** [comment]
Confirm: fixed in 0.5.4.15. Thanks!
